### PR TITLE
provision out unfilterred maf

### PR DESCRIPTION
**Pull request by gavin-peng**
for ticket https://jira.oicr.on.ca/browse/GRD-830
The changes are now 
* provision out the unfilterred tumor, normal and matched maf files
* Added "filterMafFile" boolean input to decide whether filter the maf file, so for research projects it can set to false, for clinical projects it can set to true

### CODE REVIEW

**Comment by gavin-peng** [comment]
Added one commit with two changes:
1, removed genesToKeep from the script 
2, changed output file names: matchedMaf -> somaticMaf; machtedFilteredMaf -> somaticFilteredMaf

question: also change the matchedVepVcf and index name?

**Review by lheisler — CHANGES_REQUESTED** [review]
changes to the outputs should be labels but ALSO how the files are named.
I think this is an opportunity to also introduce cleaner labels and names, as this workflow is not really in use yet for RUO data release or by CGI for reporting

suggested changes, 
no need to indicate VEP the labels, as these are ALL vep annotated (ie. it is not needed to differentiate).
the final filtered MAFs can be labelled as
tumorFilteredMaf (tumourMaf with normal annotation + filtering)
somaticFilteredMaf (matched/somaticMAF + filtering)

output file names should be simlar, using "Somatic" instead of "Matched".  its hard to figure out exactly what these will look like.  Gavin, can you indicate what extensions you expect to see on each of the output files



    File tumorVcf = variantEffectPredictor.outputVcf[0]
    File tumorVcfIndex = variantEffectPredictor.outputTbi[0]
    File normalVcf = variantEffectPredictor.outputVcf[1]
    File normalVcfIndex = variantEffectPredictor.outputTbi[1]
    File somaticVcf = matchedVep.outputVcf
    File somaticVcfIndex = matchedVep.outputTbi
    File? tumorMaf = variantEffectPredictor.outputMaf[0]
    File? normalMaf = variantEffectPredictor.outputMaf[1]
    File? somaticMaf = matchedVep.outputMaf
    File? tumorFilteredMaf = filterMaf.filteredMaf
    File? somaticFilteredMaf = matchedFilterMaf.filteredMaf

**Comment by bibilujan** [comment]
Along the lines of Larry's review, can you update the description on https://github.com/oicr-gsi/mutect2Consensus/blob/12665e01de7ac38f1698e44b02e7e517e8398bfe/mutect2Consensus.wdl#L253 to indicate it's from running mutect2 in somatic mode: 
**from**: "maf file of matched tumor and normal before filtering" **update to** "Unfiltered maf file generated from mutect2 run in somatic mode, with matched tumor and normal" 
@lheisler does that make sense? should other descriptions be updated too?

**Comment by gavin-peng** [comment]
More name and lable changes in new commit, after the change, a list of output file names and suffix (for the test run) is here:

tumorVcf
"OCT_011152_Ct_T_TS_REV_V2_Panel_VAL_REVTAR22.merged.vep.vcf.gz"

tumorVcfIndex
"OCT_011152_Ct_T_TS_REV_V2_Panel_VAL_REVTAR22.merged.vep.vcf.gz.tbi"

normalVcf
"/OCT_011152_Ly_R_TS_REV_V2_Panel_VAL_REVTAR23.merged.vep.vcf.gz"

normalVcfIndex
"OCT_011152_Ly_R_TS_REV_V2_Panel_VAL_REVTAR23.merged.vep.vcf.gz.tbi"

tumorMaf
"OCT_011152_Ct_T_TS_REV_V2_Panel_VAL_REVTAR22.merged.maf.gz"

normalMaf
"/OCT_011152_Ly_R_TS_REV_V2_Panel_VAL_REVTAR23.merged.maf.gz"

tumorfilteredMaf
"/call-filterMaf/execution/OCT_011152_Ct_T_TS_REV_V2_Panel_VAL_REVTAR22_filtered_maf.gz"

somaticVcf
"OCT_011152_Ct_T_TS_REV_V2_Panel_VAL_REVTAR22_somatic.merged.vep.vcf.gz"

somaticVcfIndex
"OCT_011152_Ct_T_TS_REV_V2_Panel_VAL_REVTAR22_somatic.merged.vep.vcf.gz.tbi"

somaticMaf
"OCT_011152_Ct_T_TS_REV_V2_Panel_VAL_REVTAR22_somatic.merged.maf.gz"

somaticFilteredMaf
"OCT_011152_Ct_T_TS_REV_V2_Panel_VAL_REVTAR22_somatic_filtered_maf.gz"

Note the "vep" suffix is genearted from the vep workflow. The "merged" suffix  is from the consenusCruncher workflow, don't know if it's clear to you, rename or remove it?

File: mutect2Consensus.wdl [hunk 1]
```diff
@@ -209,50 +213,65 @@ workflow mutect2Consensus {
...
+      tumorMaf: {
+          description: "maf file of tumor, before filtering",
+          vidarr_label: "tumorMaf"
+      },
+      normalMaf: {
+          description: "maf file of normal, before filtering",
+          vidarr_label: "normalMaf"
       },
       filteredMaf: {
           description: "maf file after filtering",
```
Comment (bibilujan) [inline mutect2Consensus.wdl]: can you update the description of filteredMaf to: "tumour Maf with normal annotation and filtering"

**Review by bibilujan — COMMENTED** [review]
minor request on file description.

File: mutect2Consensus.wdl (hunk 1 above)
Comment (gavin-peng) [reply mutect2Consensus.wdl]: thanks, updated

**Comment by lheisler** [comment]
file names look good now, not a problem to keep vep in the name if it is just being carried forward.

**Review by lheisler — APPROVED** [review]

**Review by bibilujan — APPROVED** [review]

**Pull request merged**